### ISSUE TITLE
Javascript needs ** and not ^ for exponentiation.

### ISSUE THREAD

**Issue by sangwinc** [maths/moodle-qtype_stack#814]
If we have a Maxima expression such as `x^2` and we want to embed this into javascript (e.g. JSXGraph) then we need to output `x**2`, but the block `{# ... #}` output's Maxima's format and not JSXGraph.  The new `[[jsstring]]` block (https://github.com/maths/moodle-qtype_stack/blob/dev/stack/cas/castext2/blocks/jsstring.block.php )  is not designed to do this conversion.  How do we sort this out?

**Comment by aharjula** [comment]
To clarify my earlier statements... Because a general solution is impossible, I feel strongly that there should not be a CASText level feature for converting expression syntaxes. CASText can still be used to do something like `{@maxima_to_jessiecode_string(ex)@}` if that type of a function ever exists, but those using it will need to do quite a lot of extra work to ensure that that function can deal with the expression and that they deal with the cases where it cannot.

I also feel strongly that the behaviour of `{#...#}` or `[[jsstring/]]` should not change one bit in relation to this.

**Comment by sangwinc** [comment]
Agreed, we're not going to change the behaviour of `{#...#}` or `[[jsstring/]]`.

**Comment by aharjula** [comment]
Some examples for plotting using JSXGraph if one just deals with raw student input. Note that the first one is the only one where we turn `^` to `**`, and it is also the one that I would highly recommend against. One should be able to test what breaks quite fast with this. Note that this does not include logic for adjusting the bounding box, but it should not be too difficult to just evaluate some values and pick the range for y etc...

[questions-4.0-Plöt-20220706-1341.xml.txt](https://github.com/maths/moodle-qtype_stack/files/9054049/questions-4.0-Plot-20220706-1341.xml.txt)

As one can see the JessiCode parser works quite well with Maxima already, but if you start throwing more complex expressions and functions to it it will surely break at some point. Note how that example also demonstrates how even the numerical evaluation requires additional logic if you start going complex.
